chainer/functions/loss/triplet.py

```diff
@@ -46,29 +47,37 @@ def forward(self, inputs):
             loss = xp.sum(self.dist_hinge) / N
         else:
             loss = self.dist_hinge
+
+        self.retain_inputs((0, 1, 2))
         return xp.array(loss, dtype=numpy.float32),
 
-    def backward(self, inputs, gy):
-        xp = cuda.get_array_module(*inputs)
+    def backward(self, indexes, grad_outputs):
+        anchor, positive, negative = self.get_retained_inputs()
 
-        anchor, positive, negative = inputs
         N = anchor.shape[0]
-
         x_dim = anchor.shape[1]
+
+        xp = cuda.get_array_module(anchor)
         tmp = xp.repeat(self.dist_hinge[:, None], x_dim, axis=1)
         mask = xp.array(tmp > 0, dtype=numpy.float32)
```

Comment:
We need to get `mask` (and `tmp`) in `chainer.Variable` level with `chainer.functions` instead of NumPy/CuPy level operations in `backward` method.

Reply:
Please see above comments.